### DISCUSSION TITLE
Building octave using msys2?

### DISCUSSION THREAD

**Discussion by goyalyashpal** [msys2/MSYS2-packages#3513] [Ideas]
Can octave be built using msys2?
_fantasizing:_ It would be nice to see it in this list: [who-is-using-msys2]

[who-is-using-msys2]: https://www.msys2.org/docs/who-is-using-msys2/

### Links/Build instructions

https://wiki.octave.org/Windows_Installer 
 

doing `C-f` : 
* msys : found twice
* msys2 : 0
* cygwin : 0

### Why?
* The directory structure of both is _remarkably_ same!
* It's also available in msys2-packages: [`pacman -Ss octave`][octave]
* Inspired by: https://github.com/JuliaLang/julia/issues/3640

[octave]: https://packages.msys2.org/base/mingw-w64-octave

<img src="https://user-images.githubusercontent.com/19423063/214397547-a958e1e5-8e62-4c63-b7f7-3973b70c85a7.png" width="300px"/>

### Question:
0. wait, as it's available on the pacakges, doesn't it mean that it can be build using msys2 already?

Oh, and also, if the upstream octave starts building using msys2, then will this help in:
1. reducing the size of octave install, or in
2. cleaning up that exact replica in install directory in any way?

**Comment by mmuetzel** [comment]
Are you proposing that the Octave project should stop using their MXE derivative in favor of MSYS2 for their Octave for Windows binaries?
If so, I don't think that this is something that the MSYS2 project should decide.
If you have good arguments why that would be better than their current process, you could suggest that, e.g., on their forum: 
https://octave.discourse.group/

**Reply by goyalyashpal** [reply]
Found some resources while searching for it: 🔍 : [mxe vs msys2]

* MXE Octave: Update MSYS2 packages to newer versions
https://octave.discourse.group/t/3357

* linuxtowindows_crosscompilation_using_msys2
https://www.reddit.com/r/cpp/comments/v5ctsm/

* mxe VS quasi-msys2
https://www.libhunt.com/compare-mxe-vs-quasi-msys2

[mxe vs msys2]: https://duckduckgo.com/?q=mxe+vs+msys2&t=vivaldi&ia=web
